server/src/test/java/org/elasticsearch/index/mapper/TestRuntimeField.java

```diff
@@ -61,7 +61,7 @@ public Query termQuery(Object value, QueryShardContext context) {
         @Override
         public Map<String, Parser> getRuntimeFieldTypes() {
             return Map.of(
-                "string", (name, node, parserContext) -> new TestRuntimeField(name, "string"),
+                "keyword", (name, node, parserContext) -> new TestRuntimeField(name, "keyword"),
```

Comment:
FYI - this will clash with a test I'm adding in #66030 so depending on who gets there first one of us will need to update :)

Reply:
yes this is odd, in the beginning I was only plugging in a "test" field, then a bunch of them, but now string is no longer good because dynamic templates need to know how to map a string field by default, and that default is keyword, while it is text for concrete fields.